test/Dapr.E2E.Test/Workflows/WorkflowTest.cs

```diff
@@ -11,25 +11,76 @@
 // limitations under the License.
 // ------------------------------------------------------------------------
 using System;
+using System.IO;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Dapr.Client;
 using FluentAssertions;
 using Xunit;
+using System.Linq;
+using System.Diagnostics;
 
 namespace Dapr.E2E.Test
 {
     [Obsolete]
     public partial class E2ETests
     {
+        [Fact]
+        public async Task TestWorkflowLogging()
+        {
+            // This test starts the daprclient and searches through the logfile to ensure the
+            // workflow logger is correctly logging the registered workflow(s) and activity(s)
+
+            Dictionary<string, bool> logStrings = new Dictionary<string, bool>();
+            logStrings["PlaceOrder"] = false;
+            logStrings["ShipProduct"] = false;
+            var logFilePath = "../../../../../test/Dapr.E2E.Test.App/log.txt";
+            var allLogsFound = false;
+            var timeout = 30; // 30s
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(timeout));
+            using var daprClient = new DaprClientBuilder().UseGrpcEndpoint(this.GrpcEndpoint).UseHttpEndpoint(this.HttpEndpoint).Build();
+            var health = await daprClient.CheckHealthAsync();
+            health.Should().Be(true, "DaprClient is not healthy");
+
+            var searchTask = Task.Run(() =>
+            {
+                using (StreamReader reader = new StreamReader(logFilePath))
+                {
+                    string line;
+
+                    while ((line = reader.ReadLine()) != null)
+                    {
+                        foreach (var entry in logStrings)
+                        {
+                            if (line.Contains(entry.Key))
+                            {
+                                logStrings[entry.Key] = true;
+                            }
+                        }
+                        allLogsFound = logStrings.All(k => k.Value);
+                        if (allLogsFound)
+                        {
+                            break;
+                        }
+                    }
+                }
+            }, cts.Token);
```

Comment:
AFAIK, this `CancellationToken` is only used if the `Task` has not yet already started, which it most likely has, so it's possible that the task never ends (in the case of not all expected logs being found).

You could use the `async` method `Stream.ReadLineAsync(CancellationToken)` to provide a means of abandoning the search when time expires.  That would also allow you to avoid calling `Task.WaitAll()`, which is a synchronously-blocking method.